assets/js/payment-method-extensions/payment-methods/cod/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { registerPaymentMethod } from '@woocommerce/blocks-registry';
+import { __ } from '@wordpress/i18n';
+import { getSetting } from '@woocommerce/settings';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * Internal dependencies
+ */
+import { PAYMENT_METHOD_NAME } from './constants';
+
+const settings = getSetting( 'cod_data', {} );
+const defaultLabel = __(
+	'Cash on delivery',
+	'woo-gutenberg-products-block'
+);
+const label = decodeEntities( settings.title ) || defaultLabel;
+
+/**
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+/**
+ * Content component
+ */
+const Content = () => {
+	return <div>{ decodeEntities( settings.description || '' ) }</div>;
+};
+
+/**
+ * Label component
+ *
+ * @param {*} props Props from payment API.
+ */
+const Label = ( props ) => {
+	const { PaymentMethodLabel } = props.components;
+	return <PaymentMethodLabel text={ label } />;
+};
+
+/**
+ * Determine whether COD is available for this cart/order.
+ *
+ * @param boolean cartNeedsShipping True if the cart contains any physical/shippable products.
+ * @return boolean True if COD payment method should be displayed as a payment option.
+ */
+const canMakePayment = ( { cartNeedsShipping } ) => {
```

Comment:
`Cash on delivery` can be selectively enabled for specific shipping methods only: https://docs.woocommerce.com/document/cash-on-delivery/

![Screen Capture on 2020-07-08 at 16-48-11](https://user-images.githubusercontent.com/4167300/86877379-e3109480-c13a-11ea-902d-0089495dfcda.gif)

I think we'll need to extend this interface and pass the selected shipping option (alongside cart info), and call it whenever user changes shipping. Will keep working on this tomorrow, wanted to mention this now in case we have a plan for how this would work.

cc @nerrad @Aljullu

Reply:
Good catch. I don't think anything has been discussed about how to implement that, so your proposed solution of passing the shipping option to the payment method sounds good to me. :+1:

Reply:
This is a good example of a scenario I was thinking of and [why various `cartData` is passed through the `canMakePayment` method](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/5e668c083f5c9492cfd0d4a5f9dd56a463f5b1d5/assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js#L59-L92). So 👍 for taking this approach (and noticing that it applied to cash on delivery).

Reply:
Follow up issue: #2838